### ISSUE TITLE
[Advice] is it possible to add an option `autoIntrospection: boolean` in new ApolloClient?

### ISSUE THREAD

**Issue by xialvjun** [apollographql/apollo-client#1855]
Since there is `http://dev.apollodata.com/react/initialization.html#fragment-matcher`, I think it can be done auto...

**Comment by stubailo** [comment]
Yeah we wanted to have this, but it needs a few changes to the core around initialization. Since there will be a time when introspection data is not ready yet, and we can't handle queries before then. Also, it's not clear what to do if the introspection returns an error, or the network request is lost, etc.

It's easier to do it in a step up front, but it might be nice to have a simpler way to set that up.

**Comment by xialvjun** [comment]
Thanks. Since all of our code is based on Promise, queries before introspection will waiting some time is not a problem. As to introspection returns an error, well, it indeed is a problem.

**Comment by stubailo** [comment]
> queries before introspection will waiting some time is not a problem

Yep - agreed, but it still requires some architecture changes inside the core implementation to know to wait for that.

**Comment by jbaxleyiii** [comment]
This would probably be best as a build step anyway yes? @xialvjun if this was possible at build time instead of runtime, would that meet your needs?

**Comment by xialvjun** [comment]
Well, I have indeed pushed it into the building process.
```
// package.json
"scripts": {
  "schema.json": "curl 'http://xxx/graphql' -H 'content-type: application/json' --data-binary '{\"query\":\"{__schema{types{kind name possibleTypes{name}}}}\",\"variables\":null,\"operationName\":null}' --compressed > schema.json",
  "build": "npm run schema.json & webpack --optimize-minimize",
}
```

I don't know which is the best way.
